nautobot/extras/jobs.py

```diff
@@ -123,26 +131,21 @@ def __call__(self, *args, **kwargs):
         self.active_test = "run"
         context_class = JobHookChangeContext if isinstance(self, JobHookReceiver) else JobChangeContext
         change_context = context_class(user=self.user, context_detail=self.class_path)
-        # TODO: re-add profiling feature
-        # if profile:
-        #     import cProfile
-
-        #     # TODO: This should probably be available as a file download rather than dumped to the hard drive.
-        #     # Pending this: https://github.com/nautobot/nautobot/issues/3352
-        #     profiling_path = f"/tmp/nautobot-jobresult-{self.request.id}.pstats"
-
-        #     # TODO: Context manager for this is added in 3.8
-        #     profiler = cProfile.Profile()
-        #     profiler.enable()
-        #     output = job.run(data=data, commit=commit)
-        #     profiler.disable()
-        #     profiler.dump_stats(profiling_path)
-        #     job.log_info(obj=None, message=f"Wrote profiling information to {profiling_path}.")
-        # else:
-        #     output = job.run(data=data, commit=commit)
 
         with change_logging(change_context):
-            return self.run(*args, **deserialized_kwargs)
+            if self.celery_kwargs.get("nautobot_job_profile", False) is True:
```

Comment:
Is there a case where we want it to only be explicitly `True` and not just evaluate to `True`?
```suggestion
            if self.celery_kwargs.get("nautobot_job_profile", False):
```

Reply:
Unless the serializer is broken we shouldn't need to. If it was being hard coded to True from an environment variable, it should be coerced to a python boolean by the python code that's grabbing the environment variable.